frontend/src/employee-frontend/components/messages/MessageContext.tsx

```diff
@@ -501,8 +512,20 @@ export const MessageContextProvider = React.memo(
       },
       [setSelectedThread]
     )
-    const reply = useRestApi(replyToThread, setReplyResponse)
-    const sendReply = useCallback(reply, [reply])
+    const sendReply = useRestApi(
+      (params: ReplyToThreadParams) =>
+        replyToThread(params).then((result) => {
+          if (result.isSuccess) {
+            setCancelableMessage({
+              accountId: params.accountId,
+              messageId: result.value.message.id,
+              sentAt: HelsinkiDateTime.now()
+            })
+          }
+          return result
+        }),
+      setReplyResponse
+    )
```

Comment:
It would be so cool if we had a chained async calls abstraction in the frontend, because this is quite ugly and has been used in many places 😞